<!-- An online display running on real-time to show the state of the board -->
<launch>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find baxter_tictactoe)/custom_rosconsole.conf"/>

    <include file="$(find baxter_tictactoe)/launch/usb_cam_node.launch" />

    <!-- Range thresholds for the red tokens -->
    <rosparam param = "baxter_tictactoe/hsv_red">
        H: [160,  20]
        S: [ 40, 196]
        V: [ 50, 196]
    </rosparam>

    <!-- Range thresholds for the blue tokens -->
    <rosparam param = "baxter_tictactoe/hsv_blue">
        H: [ 90, 130]
        S: [ 70, 256]
        V: [ 70, 256]
    </rosparam>

    <param name="baxter_tictactoe/yale_logo_file" value="$(find baxter_tictactoe)/img/yale_logo.png"/>

    <!-- If show is set to true, then the board state sensing will show the board with the hsv-color filtering for red and blue.-->
    <arg name="show" default="false" />

    <!-- Minimum area size for which a cell is considered colored. -->
    <!-- It depends on the distance between camera and board, and camera resolution -->
    <param name="baxter_tictactoe/area_threshold" type="int" value="650" />

    <node name="board_state_sensor" pkg="baxter_tictactoe" type="board_state_sensor" args="--show $(arg show)" respawn="false" output="screen" required="false">
        <remap from="/baxter_tictactoe/image" to="/usb_cam/image_raw"/>
    </node>

    <node name="baxterDisplay" pkg="baxter_tictactoe" type="baxterDisplay" respawn="false" output="screen">
        <remap from="baxter_display" to="/robot/xdisplay"/>
        <remap from="board_state"    to="/baxter_tictactoe/board_state"/>
    </node>

    <!-- <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/robot/xdisplay"/>
        <param name="autosize" value="true" />
    </node> -->

    <node name="camera_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/baxter_tictactoe/board_state_img"/>
        <param name="autosize" value="true" />
    </node>

</launch>
